### PR TITLE
fix: enum discriminant overflow

### PR DESCRIPTION
**Pull request by Salka1988**
### Checklist
- [x] I have linked to any relevant issues.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels-macros/src/derive/utils.rs [hunk 1]
```diff
@@ -75,15 +75,12 @@ pub(crate) fn extract_variants(
...
-                let current_discriminant = discriminant;
-                discriminant += 1;
-
                 Ok(ExtractedVariant::Normal {
                     info: VariantInfo {
                         name: variant.ident,
                         is_unit,
                     },
-                    discriminant: current_discriminant,
+                    discriminant: discriminant as u8,
```
Comment (hal3e) [inline packages/fuels-macros/src/derive/utils.rs:83]: Even though, sway uses `u8` for the discriminant, I would still guard against this conversion with an `try_into`. Someone could manually make an `Enum` or a json abi with more than 256 variants.

```rust
  let discriminant = discriminant.try_into().map_err(|_| {
      Error::new_spanned(&variant.ident, "Enums can not have more than 256 variants")
  })?;

  Ok(ExtractedVariant::Normal {
      info: VariantInfo {
          name: variant.ident,
          is_unit,
      },
      discriminant,
  })
```

We could also write an UI test for this.

**Review by hal3e — CHANGES_REQUESTED** [review]
Nice!

File: packages/fuels-macros/src/derive/utils.rs (hunk 1 above)
Comment (Salka1988) [reply packages/fuels-macros/src/derive/utils.rs:83]: You're right, I'll do it.

**Review by MujkicA — APPROVED** [review]

**Review by hal3e — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
Good job @Salka1988

**Pull request merged**